docs/designs/update-strategy.md

```diff
@@ -0,0 +1,661 @@
+├─ UpdateStrategy (controls PCS replica updates)
+│  ├─ RollingUpdate: one replica at a time
+│  └─ ReplicaRecreate: tear down entire replica
+│
+├─ PCS Replica 0
+│  ├─ Standalone PodCliques (concurrent updates)
+│  │  └─ frontend: UpdateStrategy (controls pod updates)
+│  │     └─ maxUnavailable/maxSurge
+│  │
+│  └─ PodCliqueScalingGroups (concurrent updates)
+│     ├─ prefill: UpdateStrategy (controls PCSG replica updates)
+│     │  └─ maxUnavailable/maxSurge
+│     └─ decode: UpdateStrategy (controls PCSG replica updates)
+│        └─ maxUnavailable/maxSurge
+│
+└─ PCS Replica 1, 2, ... (sequential per PCS strategy)
+```
+
+## API Structure
+
+### 1. PodCliqueSetSpec.UpdateStrategy
+
+Controls **how PCS replicas are updated** (the top-most level).
+
+```go
+type PodCliqueSetUpdateStrategy struct {
+    Type          PodCliqueSetUpdateStrategyType
+    RollingUpdate *PodCliqueSetRollingUpdateStrategy
+}
+
+type PodCliqueSetUpdateStrategyType string
+const (
+    RollingUpdate    // Update replicas sequentially
+    ReplicaRecreate  // Delete/recreate entire replica
+)
+
+type PodCliqueSetRollingUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many PCS replicas can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra PCS replicas to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `RollingUpdate` with `maxUnavailable=1, maxSurge=0`
+
+### 2. PodCliqueTemplateSpec.UpdateStrategy
+
+Controls **how pods update within a standalone PodClique**.
+
+```go
+type ComponentUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many pods can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra pods to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `maxUnavailable=1, maxSurge=0`
+**Scope:** Only applies to standalone PodCliques (not in a PCSG)
+
+### 3. PodCliqueScalingGroupConfig.UpdateStrategy
+
+Controls **how PCSG replicas update within a scaling group**.
+
+```go
+type ComponentUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many PCSG replicas can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra PCSG replicas to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `maxUnavailable=1, maxSurge=0`
+**Scope:** Controls the update of PCSG replicas (groups of PodCliques)
+
+## Update Behavior
+
+### RollingUpdate (Default)
+
+The default RollingUpdate behavior is described in the [Motivation](#motivation) section. When using the RollingUpdate strategy, `maxUnavailable` and `maxSurge` settings at the PodCliqueSet level are invalid and will be rejected by the validation webhook - PCS replicas are always updated one at a time sequentially. However, PC and PCSG `updateStrategy` settings (maxUnavailable/maxSurge) are observed and control how pods and PCSG replicas update within each PCS replica.
```

Comment:
I'm bit confused by this section.  PCSGs always get recreated right now. We can use maxUnavailable/maxSurge to define how that happens more precisely.